pkg/search/postgres/store.go

```diff
@@ -295,7 +295,17 @@ func (s *GenericStore[T, PT]) DeleteMany(ctx context.Context, identifiers []stri
 
 	// Batch the deletes
 	localBatchSize := deleteBatchSize
-	numRecordsToDelete := len(identifiers)
+	var err error
+	var tx *postgres.Tx
+	if !postgres.HasTxInContext(ctx) {
```

Comment:
What are the possible cases with the transaction in context ?
Is it removed from context on commit or rollback ?
From what I read here, I understand it is supposed that the transaction from context is active (after begin but before commit or rollback).

Reply:
If you put a transaction in a context, it will be able to span across multiple stores and do everything in a txn

Reply:
It's assumed that the transaction in the context is active